integration_tests/juniper_tests/src/issue_922.rs

```diff
@@ -0,0 +1,121 @@
+use juniper::*;
+
+struct Query;
+
+#[juniper::graphql_object]
+impl Query {
+    fn characters(executor: &Executor) -> Vec<CharacterValue> {
+        executor.look_ahead();
+
+        vec![
+            Into::into(Human {
+                id: 0,
+                name: "human-32".to_owned(),
+            }),
+            Into::into(Droid {
+                id: 1,
+                name: "R2-D2".to_owned(),
+            }),
+        ]
+    }
+}
+
+#[juniper::graphql_interface(for = [Human, Droid])]
+trait Character {
+    fn id(&self) -> i32;
+
+    fn name(&self) -> String;
+}
+
+#[derive(juniper::GraphQLObject)]
+#[graphql(impl = CharacterValue)]
+struct Human {
+    pub id: i32,
+    pub name: String,
+}
+
+#[juniper::graphql_interface]
+impl Character for Human {
+    fn id(&self) -> i32 {
+        self.id
+    }
+
+    fn name(&self) -> String {
+        self.name.clone()
+    }
+}
+
+#[derive(juniper::GraphQLObject)]
+#[graphql(impl = CharacterValue)]
+struct Droid {
+    pub id: i32,
+    pub name: String,
+}
+
+#[juniper::graphql_interface]
+impl Character for Droid {
+    fn id(&self) -> i32 {
+        self.id
+    }
+
+    fn name(&self) -> String {
+        self.name.clone()
+    }
+}
+
+type Schema = juniper::RootNode<'static, Query, EmptyMutation<()>, EmptySubscription<()>>;
+
+#[tokio::test]
+async fn test_fragment_on_interface() {
+    let query = r#"
+        query Query {
+            characters {
+                ...CharacterFragment
+            }
+        }
+
+        fragment CharacterFragment on Character {
+            __typename
+            ... on Human {
+                id
+                name
+            }
+            ... on Droid {
+                id
+                name
+            }
+        }
+    "#;
+
+    let (res, errors) = juniper::execute(
```

Comment:
@ArsileLuci consider to cover both `execute` (async) and `execute_sync` in tests.